pacman/utilities/algorithm_utilities/placer_algorithm_utilities.py

```diff
@@ -63,50 +64,67 @@ def group_vertices(vertices, same_group_as_function):
         A dictionary of vertex to list of vertices that are grouped with it
     """
 
-    # Dict of vertex to list of vertices on same chip (repeated lists expected)
+    groups = create_vertices_groups(vertices, same_group_as_function)
+    # Dict of vertex to setof vertices on same chip (repeated lists expected)
+    # A empty set value indicates a set that is too big.
     same_chip_vertices = dict()
+    for group in groups:
+        for vertex in group:
+            same_chip_vertices[vertex] = group
+    for vertex in vertices:
+        if vertex not in same_chip_vertices:
+            same_chip_vertices[vertex] = {vertex}
+    return same_chip_vertices
+
+
+def add_set(all_sets, new_set):
+    for a_set in all_sets:
+        intersection = new_set & a_set
+        if intersection:
+            all_sets.remove(a_set)
+            union = a_set | new_set
+            add_set(all_sets, union)
```

Comment:
There should be some way to avoid writing this recursively.

Reply:
Indeed, I think there ought to be a way to write this using a single pass.

Reply:
Yep Done

Reply:
Looks good